pkg/resourceinterpreter/configurableinterpreter/configmanager/manager.go

```diff
@@ -45,8 +45,7 @@ func (configManager *interpreterConfigManager) HasSynced() bool {
 	if configManager.initialSynced.Load().(bool) {
 		return true
 	}
-
-	if configManager.HasSynced() {
+	if configuration, err := configManager.lister.List(labels.Everything()); err == nil && len(configuration) == 0 {
```

Comment:
`len(configuration) == 0` means synced?

Reply:
Maybe shall use `informer.IsInformerSynced()`

Reply:
The relevant comments can be described here.

Reply:
> `len(configuration) == 0` means synced?
https://github.com/karmada-io/karmada/blob/d3b19ed744529e95d210bbc4e4b113c88b4d6b31/pkg/resourceinterpreter/customizedinterpreter/configmanager/manager.go#L50-L58

Reply:
But if informer is not synced, the `lister.Lister` also returns `[]` items and `nil` error. I test it as below:
```go
func TestName2(t *testing.T) {
        // no server servers on this address
	config, _ := clientcmd.BuildConfigFromFlags("http://127.0.0.1", "")
	clientSet := kubernetes.NewForConfigOrDie(config)
	factory := informers.NewSharedInformerFactory(clientSet, 0)
	lister := factory.Core().V1().Nodes().Lister()

	stopCh := make(chan struct{})
	defer close(stopCh)
	factory.Start(stopCh)
	go factory.WaitForCacheSync(stopCh)

	ticker := time.NewTicker(time.Second)
	for range ticker.C {
		nodes, err := lister.List(labels.Everything())
		t.Log(nodes, err)
	}
}
```

And output:
```
W1116 11:07:20.838592   95036 reflector.go:324] pkg/mod/k8s.io/client-go@v0.23.5/tools/cache/reflector.go:167: failed to list *v1.Node: Get "http://127.0.0.1/api/v1/nodes?limit=500&resourceVersion=0": dial tcp 127.0.0.1:80: connect: connection refused
E1116 11:07:20.838731   95036 reflector.go:138] pkg/mod/k8s.io/client-go@v0.23.5/tools/cache/reflector.go:167: Failed to watch *v1.Node: failed to list *v1.Node: Get "http://127.0.0.1/api/v1/nodes?limit=500&resourceVersion=0": dial tcp 127.0.0.1:80: connect: connection refused
    cli_test.go:62: [] <nil>
W1116 11:07:22.127046   95036 reflector.go:324] pkg/mod/k8s.io/client-go@v0.23.5/tools/cache/reflector.go:167: failed to list *v1.Node: Get "http://127.0.0.1/api/v1/nodes?limit=500&resourceVersion=0": dial tcp 127.0.0.1:80: connect: connection refused
E1116 11:07:22.127648   95036 reflector.go:138] pkg/mod/k8s.io/client-go@v0.23.5/tools/cache/reflector.go:167: Failed to watch *v1.Node: failed to list *v1.Node: Get "http://127.0.0.1/api/v1/nodes?limit=500&resourceVersion=0": dial tcp 127.0.0.1:80: connect: connection refused
    cli_test.go:62: [] <nil>
    cli_test.go:62: [] <nil>
    cli_test.go:62: [] <nil>
W1116 11:07:25.236868   95036 reflector.go:324] pkg/mod/k8s.io/client-go@v0.23.5/tools/cache/reflector.go:167: failed to list *v1.Node: Get "http://127.0.0.1/api/v1/nodes?limit=500&resourceVersion=0": dial tcp 127.0.0.1:80: connect: connection refused
E1116 11:07:25.236924   95036 reflector.go:138] pkg/mod/k8s.io/client-go@v0.23.5/tools/cache/reflector.go:167: Failed to watch *v1.Node: failed to list *v1.Node: Get "http://127.0.0.1/api/v1/nodes?limit=500&resourceVersion=0": dial tcp 127.0.0.1:80: connect: connection refused
    cli_test.go:62: [] <nil>
    cli_test.go:62: [] <nil>
    cli_test.go:62: [] <nil>
    cli_test.go:62: [] <nil>
    cli_test.go:62: [] <nil>
W1116 11:07:30.568426   95036 reflector.go:324] pkg/mod/k8s.io/client-go@v0.23.5/tools/cache/reflector.go:167: failed to list *v1.Node: Get "http://127.0.0.1/api/v1/nodes?limit=500&resourceVersion=0": dial tcp 127.0.0.1:80: connect: connection refused
E1116 11:07:30.568490   95036 reflector.go:138] pkg/mod/k8s.io/client-go@v0.23.5/tools/cache/reflector.go:167: Failed to watch *v1.Node: failed to list *v1.Node: Get "http://127.0.0.1/api/v1/nodes?limit=500&resourceVersion=0": dial tcp 127.0.0.1:80: connect: connection refused
    cli_test.go:62: [] <nil>
```

We can see, informer is always not synced, but lister returns `[]` and `nil`

Reply:
Just as the comment said, `the empty list we initially stored is valid to use. ` There maybe not a problem?

Reply:
Another word, when this function will return false?

Reply:
When an error occurs in the list, the probability is low.

Reply:
I agree with @ikaven1024, maybe we can hold the informer into `interpreterConfigManager` , not just the lister.
So, that we can check the sync state by `IsInformerSynced` or `WaitForCacheSync`.

> 		// the empty list we initially stored is valid to use.
>		// Setting initialSynced to true, so subsequent checks
>		// would be able to take the fast path on the atomic boolean in a
>		// cluster without any customization configured.

By the way, I think the comments didn't answer @ikaven1024's question.

Reply:
Let's do it by the next PR.